client/lib/signup/test/signup/config/flows.js

```diff
@@ -25,8 +25,6 @@ var flows = {
 };
 
 module.exports = {
-	currentFlowName: 'simple_flow',
```

Comment:
Right, I think I managed to gather that after a lot of reading. :smile: One thing that confused me is what `currentFlowName` was supposed to be. My best guess after following the code trail is that it is an artifact of an old way of dealing with flows, but which had taken on a new purpose (replacing `defaultFlowName`).

This PR includes tests to cover every use I could think of for `getFlowName` (which doesn't mean it's right!) and which hopefully will make the naming more clear. Do you have any suggestions?